Using volumes flag for mounting causes issue:  Root folder could not be created. Path: /rootBucket in v2.5.0 and v2.5.1

### Problem

#### TL;DR
* Unable to use volume mount with releases `2.5.0` and `2.5.1`

#### Here is my simple setup: 
```
.
├── docker-compose.yaml
└── tobemounted
    └── test.yaml

1 directory, 2 files
```

#### Here are the contents of my `docker-compose.yaml`: 

```
version: "3.9"
services:
  s3mock:
    image: adobe/s3mock
    environment:
      initialBuckets: testbucket
      root: rootBucket
    volumes:
      - ./tobemounted:/rootBucket/testbucket/
```

#### Expected
* Container spins up as expected 

#### Actual
```
test-s3mock-1  | Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [java.io.File]: Factory method 'rootFolder' threw exception; nested exception is java.lang.IllegalStateException: Root folder could not be created. Path: /rootBucket
```

### Additional Info
* Interestingly this seems to be happening only with `2.5.x` releases and not with `2.4.x`, for example if i alter my `docker-compose.yaml` to look like below, it works as expected 

```
version: "3.9"
services:
  s3mock:
    image: adobe/s3mock:2.4.16
    environment:
      initialBuckets: testbucket
      root: rootBucket
    volumes:
      - ./tobemounted:/rootBucket/testbucket/
```

It _seems_ to be a valid issue, but I am also new to this repo, so I may be doing something horrendously wrong as well. Glad to share more info (ex: enable debug/traces and provide with output etc).

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,9 +8,13 @@
 * Version updates
   * TBD
 
+## 2.5.2
+* Features and fixes
+  * Correctly detect and use existing root folder (fixes #786)
+
 ## 2.5.1
 * Features and fixes
-  * Use @SpringBootApplication to configure S3Mock (fixes #782)
+  * Use `@SpringBootApplication` to configure S3Mock (fixes #782)
 
 ## 2.5.0
 * Features and fixes

--- a/server/src/main/java/com/adobe/testing/s3mock/store/StoreConfiguration.java
+++ b/server/src/main/java/com/adobe/testing/s3mock/store/StoreConfiguration.java
@@ -77,13 +77,16 @@ class StoreConfiguration {
     if (!properties.isRetainFilesOnExit()) {
       root.deleteOnExit();
     }
-    if (!root.mkdir()) {
+    if (root.exists()) {
+      LOG.info("Using existing folder \"{}\" as root folder. Will retain files on exit: {}",
+          root.getAbsolutePath(), properties.isRetainFilesOnExit());
+    } else if (!root.mkdir()) {
       throw new IllegalStateException("Root folder could not be created. Path: "
           + root.getAbsolutePath());
+    } else {
+      LOG.info("Successfully created \"{}\" as root folder. Will retain files on exit: {}",
+          root.getAbsolutePath(), properties.isRetainFilesOnExit());
     }
-
-    LOG.info("Using \"{}\" as root folder. Will retain files on exit: {}",
-        root.getAbsolutePath(), properties.isRetainFilesOnExit());
 
     return root;
   }

